src/python/pants/notes/2.3.x.md

```diff
@@ -0,0 +1,39 @@
+## 2.3.0.dev0 (Jan 10, 2021)
+
+### User API Changes
+
+* Remove support for pex_binary sources. ([#11445](https://github.com/pantsbuild/pants/pull/11445))
+
+* Remove custom targets for Pylint and MyPy. ([#11441](https://github.com/pantsbuild/pants/pull/11441))
+
+* Remove deprecated --pantsd-pailgun-quit-timeout option. ([#11440](https://github.com/pantsbuild/pants/pull/11440))
+
+* Remove DeprecatedRunTracker. ([#11439](https://github.com/pantsbuild/pants/pull/11439))
+
+* Remove deprecated PythonAwsLambda fields. ([#11438](https://github.com/pantsbuild/pants/pull/11438))
+
+* Remove deprecated v1 Subsystem facilities ([#11424](https://github.com/pantsbuild/pants/pull/11424))
+
+* Remove python workunits ([#11416](https://github.com/pantsbuild/pants/pull/11416))
```

Comment:
Internal only, from the PR description:

> This is a pure refactoring change that should have no user-facing effects.

Reply:
Nack - the removed code was maked API public.

Reply:
Okay, then it should go into Plugin API Change

Reply:
Agreed the API public thing is a bit of a legacy mess and a not so great system, but we document it in our public contract currently.

Reply:
Ah yes, will fix.